<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CrystalDisplay</class>
 <widget class="QWidget" name="CrystalDisplay">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>200</width>
    <height>281</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>200</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Crystal</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../icons/icons.qrc">
    <normaloff>:/Crystal.png</normaloff>:/Crystal.png</iconset>
  </property>
  <layout class="QGridLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="1" column="0" colspan="3">
    <layout class="QHBoxLayout">
     <item>
      <widget class="QLabel" name="label_Spacegroup">
       <property name="text">
        <string>Spacegroup</string>
       </property>
       <property name="buddy">
        <cstring>Spacegroup</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Spacegroup"/>
     </item>
    </layout>
   </item>
   <item row="2" column="0" colspan="3">
    <layout class="QHBoxLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <item>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="Label_A">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>A</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeA</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeA">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
         </property>
         <property name="prefix">
          <string/>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>75.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>5.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="Label_Alpha">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Alpha</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeAlpha</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeAlpha">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>179.900000000000006</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>90.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="label_B">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>B</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeB</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeB">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>75.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>5.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_Beta">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Beta</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeBeta</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeBeta">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>179.900000000000006</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>90.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="label_C">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>C</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeC</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeC">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>75.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>5.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_Gamma">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Gamma</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>latticeGamma</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NumberEdit" name="latticeGamma">
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::NoButtons</enum>
         </property>
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>179.900000000000006</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>90.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="3" column="0" colspan="3">
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="3">
    <widget class="QLabel" name="label_OMat">
     <property name="text">
      <string>orientation matrix</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="3">
    <widget class="QTableWidget" name="orientationMatrix">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="autoScroll">
      <bool>false</bool>
     </property>
     <property name="showDropIndicator" stdset="0">
      <bool>false</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::SolidLine</enum>
     </property>
     <property name="rowCount">
      <number>3</number>
     </property>
     <property name="columnCount">
      <number>3</number>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <row/>
     <row/>
     <row/>
     <column/>
     <column/>
     <column/>
    </widget>
   </item>
   <item row="6" column="0">
    <layout class="QVBoxLayout">
     <item>
      <widget class="QLabel" name="label_Omega">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Omega</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="buddy">
        <cstring>rotationOmega</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="NumberEdit" name="rotationOmega">
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-360.000000000000000</double>
       </property>
       <property name="maximum">
        <double>360.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="1">
    <layout class="QVBoxLayout">
     <item>
      <widget class="QLabel" name="label_Chi">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Chi</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="buddy">
        <cstring>rotationChi</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="NumberEdit" name="rotationChi">
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-360.000000000000000</double>
       </property>
       <property name="maximum">
        <double>360.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="2">
    <layout class="QVBoxLayout">
     <item>
      <widget class="QLabel" name="label_Phi">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Phi</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="buddy">
        <cstring>rotationPhi</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="NumberEdit" name="rotationPhi">
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-360.000000000000000</double>
       </property>
       <property name="maximum">
        <double>360.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="7" column="0" colspan="3">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="3">
    <widget class="QFrame" name="frame">
     <layout class="QHBoxLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="ToolBar" name="tollBar" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="dragStart">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="../icons/icons.qrc">:/Crystal.png</pixmap>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NumberEdit</class>
   <extends>QDoubleSpinBox</extends>
   <header>tools/numberedit.h</header>
  </customwidget>
  <customwidget>
   <class>ToolBar</class>
   <extends>QWidget</extends>
   <header>defs.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>Spacegroup</tabstop>
  <tabstop>latticeA</tabstop>
  <tabstop>latticeB</tabstop>
  <tabstop>latticeC</tabstop>
  <tabstop>latticeAlpha</tabstop>
  <tabstop>latticeBeta</tabstop>
  <tabstop>latticeGamma</tabstop>
  <tabstop>orientationMatrix</tabstop>
  <tabstop>rotationOmega</tabstop>
  <tabstop>rotationChi</tabstop>
  <tabstop>rotationPhi</tabstop>
 </tabstops>
 <resources>
  <include location="../icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>rotationOmega</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>25</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>235</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotationChi</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>92</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotationPhi</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>68</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Spacegroup</sender>
   <signal>textChanged(QString)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotSpacegroupChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeGamma</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeBeta</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>134</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeAlpha</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeC</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeB</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>latticeA</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CrystalDisplay</receiver>
   <slot>slotCellChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>36</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotCellChanged()</slot>
  <slot>slotRotationChanged()</slot>
  <slot>slotSpacegroupChanged(QString)</slot>
 </slots>
</ui>
